CMakeLists.txt

```diff
@@ -51,19 +51,34 @@ add_custom_target(Download)
 add_custom_target(fletch-build-install)
 
 # Options to control GPU support
-option(fletch_BUILD_WITH_CUDA "Build with CUDA support" FALSE)
+option(fletch_ENABLE_CUDA "Build with CUDA support" FALSE)
 
-if (fletch_BUILD_WITH_CUDA)
+if (fletch_ENABLE_CUDA)
   find_package( CUDA QUIET REQUIRED )
 
-  option(fletch_BUILD_WITH_CUDNN "Build with CUDNN support" FALSE)
-
-  if (fletch_BUILD_WITH_CUDNN)
-    set( CUDNN_ROOT "" CACHE PATH "CUDNN root folder" )
-    mark_as_advanced( CUDNN_ROOT )
+  option(fletch_ENABLE_CUDNN "Build with CUDNN support" FALSE)
+  if (fletch_ENABLE_CUDNN)
+    set( CUDNN_ROOT_DIR "" CACHE PATH "CUDNN root folder" )
+    mark_as_advanced( CUDNN_ROOT_DIR )
+    
+    find_library( CUDNN_LIBRARIES
+      NAMES cudnn libcudnn
+      HINTS ${CUDA_TOOLKIT_ROOT_DIR}
```

Comment:
Sorry I didn't mention this before, and it's not really a huge deal but ... the typical pattern for these find_package options is, if the user asked for it to be enabled, you add a REQUIRED to the find and let CMake handle the error part. It simplifies the code and makes it much smaller/easier to read. CMake will just fatal error if everything isn't correctly found